### PR TITLE
Cleanup update

### PR DESCRIPTION
**Pull request by drew-gross**
Move mongo specific object deletion logic into mongo adapter.

### CODE REVIEW

File: src/Controllers/DatabaseController.js [hunk 1]
```diff
@@ -283,50 +285,42 @@ DatabaseController.prototype.removeRelation = function(key, fromClassName, fromI
 //         one of the provided strings must provide the caller with
 //         write permissions.
 DatabaseController.prototype.destroy = function(className, query, options = {}) {
-  var isMaster = !('acl' in options);
-  var aclGroup = options.acl || [];
+  const isMaster = !('acl' in options);
```
Comment (flovilmart) [inline src/Controllers/DatabaseController.js]: we could do the same rework with replacing options by `{ acl } = {}`

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js [hunk 2]
```diff
@@ -159,27 +159,38 @@ export class MongoStorageAdapter {
...
-  deleteObject(className, objectId) {
+  // Remove all objects that match the given parse query. Parse Query should be in Parse Format.
+  // If no objects match, reject with OBJECT_NOT_FOUND. If objects are found and deleted, resolve with nothing.
+  // If there is some other error, reject with INTERNAL_SERVER_ERROR.
+
+  // Currently accepts the acl, schemaController, validate
+  // for lecacy reasons, Parse Server should later integrate acl into the query. Database adapters
+  // shouldn't know about acl.
+
+  // Currently resolves with empty object because thats what HooksController expects, but that should change.
```
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: I can make a quick PR on the HooksController if you want

Comment (drew-gross) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: If you want, sure. I think the best behaviour is probably to resolve with undefined.

Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: resolve to what you need to, I'll override the resolution in the HooksController

Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: #1595

**Pull request merged**